neonvm/controllers/virtualmachine_controller.go

```diff
@@ -444,6 +448,15 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 					Status:  metav1.ConditionTrue,
 					Reason:  "Reconciling",
 					Message: fmt.Sprintf("Pod (%s) for VirtualMachine (%s) created successfully", virtualmachine.Status.PodName, virtualmachine.Name)})
+			{
+				now := time.Now()
+				d := now.Sub(vmRunner.CreationTimestamp.Time)
+				r.Metrics.runnerCreationToVMRunningTime.Observe(d.Seconds())
+				if !virtualmachine.HasRestarted() {
+					d := now.Sub(virtualmachine.CreationTimestamp.Time)
+					r.Metrics.vmCreationToVMRunningTime.Observe(d.Seconds())
+				}
+			}
```

Comment:
Only things I'd note are:
1. This block could use a comment noting what we're doing (short one-line sorta thing)
2. IMO it'd be good to log the duration here, so that we can _in theory_ dig into the logs to figure out why metrics are weird. I'm imagining something like "we keep seeing the occasional VM that takes +inf time here [due to a bug in the state machine], how can we debug it?"

Reply:
Makes sense. Fixed in c7654bb64d275c857d62acb8332b58c76a203f34